### ISSUE TITLE
desktop connects even if "pause all synchronization"

### ISSUE THREAD

**Issue by r2evans** [nextcloud/desktop#692]
### Expected behaviour

When "pause all synchronization" is selected, no syncing connections are attempted.

### Actual behaviour

When I have already selected "pause all synchronization" and connect to (say) a hotel wifi that is intercepting my connections, Nextcloud client complains that the SSL certificates have changed.

Note: I'm not concerned about NC complaining about it in general -- if I were trying to sync and it notices the certs are different, it *should* notify me. The problem here is that it *tried* to connect.

### Steps to reproduce
1. Select "pause all synchronization"
2. Connect to a wifi that requires authentication before allowing default routing; it is important that the hotspot (hotel or cafe) intercept http and https, perhaps even responding with a simple page, before the user has authenticated such. (In my use, accessing https once the hotspot auth is complete will work without problem, meaning the https interception is just to prevent unauthorized use.
3. The NC desktop client will complain that the SSL certificate has changed.

### Client configuration
Client version: 2.3.3 build 1

Operating system: win10_64

OS language: English

Qt version used by client package (Linux only, see also Settings dialog): n/a

Client package (From ownCloud or distro) (Linux only): n/a

Installation path of client: `C:\Program Files (x86)\Nextcloud`

### Server configuration

N/A, since all behavior is normal once hotspot authentication is complete. (The server does not see the connection attempt that triggered the SSL-change error.)

### Logs

(I'll try to edit the issue later with something, getting the log to save is crashing my nextcloud client, perhaps another issue.)

**Comment by r2evans** [comment]
Repeated today (same results). (To be clear, I have two different nextcloud instances, so you'll see two sets of entries here.)

<details>
<summary>Log file</summary>

```
09-25 16:31:17:734 30084 unknown: QString::arg: Argument missing: "https://docs.nextcloud.com/desktop/2.2/" , 2
09-25 16:31:17:760 30084 unknown: QString::arg: Argument missing: "https://docs.nextcloud.com/desktop/2.2/" , 3
09-25 16:31:21:221 30084 unknown: QString::arg: Argument missing: "https://docs.nextcloud.com/desktop/2.2/" , 2
09-25 16:31:21:223 30084 unknown: QString::arg: Argument missing: "https://docs.nextcloud.com/desktop/2.2/" , 3
09-25 16:31:22:406 30084 OCC::Folder::saveToSettings: Saved folder "1" to settings, status 0
09-25 16:31:22:409 30084 OCC::SocketListener::sendMessage: SocketApi:  Sending message:  "UPDATE_VIEW:C:\\Users\\me\\oc\\dir1"
09-25 16:31:22:412 30084 OCC::SocketListener::sendMessage: SocketApi:  Sending message:  "UPDATE_VIEW:C:\\Users\\me\\oc\\dir1"
09-25 16:31:22:427 30084 OCC::ownCloudGui::slotSyncStateChange: Sync state changed for folder  "https://my.domain1.com/oc/remote.php/webdav/" :  "Sync Paused"
09-25 16:31:22:429 30084 OCC::Folder::slotTerminateSync: folder  "1"  Terminating!
09-25 16:31:22:462 30084 OCC::Folder::saveToSettings: Saved folder "2" to settings, status 0
09-25 16:31:22:464 30084 OCC::SocketListener::sendMessage: SocketApi:  Sending message:  "UPDATE_VIEW:C:\\Users\\me\\oc\\dir2"
09-25 16:31:22:466 30084 OCC::SocketListener::sendMessage: SocketApi:  Sending message:  "UPDATE_VIEW:C:\\Users\\me\\oc\\dir2"
09-25 16:31:22:494 30084 OCC::ownCloudGui::slotSyncStateChange: Sync state changed for folder  "https://my.domain2.com/cloud/remote.php/webdav/" :  "Sync Paused"
09-25 16:31:22:496 30084 OCC::Folder::slotTerminateSync: folder  "2"  Terminating!
09-25 16:31:22:783 30084 OCC::Account::resetNetworkAccessManager: Resetting QNAM
09-25 16:31:22:786 30084 OCC::ConnectionValidator::checkServerAndAuth: Checking server and authentication
09-25 16:31:22:789 30084 OCC::ConnectionValidator::checkServerAndAuth: Trying to look up system proxy
09-25 16:31:22:791 30084 OCC::Account::resetNetworkAccessManager: Resetting QNAM
09-25 16:31:22:793 30084 OCC::ConnectionValidator::checkServerAndAuth: Checking server and authentication
09-25 16:31:22:796 30084 OCC::ConnectionValidator::checkServerAndAuth: Trying to look up system proxy
09-25 16:31:22:804 30084 OCC::ConnectionValidator::systemProxyLookupDone: No system proxy set by OS
09-25 16:31:22:806 30084 OCC::AbstractNetworkJob::start: !!! OCC::CheckServerJob created for "https://my.domain1.com/oc" + "status.php" "OCC::ConnectionValidator"
09-25 16:31:22:809 30084 OCC::ConnectionValidator::systemProxyLookupDone: No system proxy set by OS
09-25 16:31:22:812 30084 OCC::AbstractNetworkJob::start: !!! OCC::CheckServerJob created for "https://my.domain2.com/cloud" + "status.php" "OCC::ConnectionValidator"
09-25 16:31:40:418 30084 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(OperationCanceledError) "Operation canceled" QVariant(Invalid)
09-25 16:31:40:419 30084 unknown: QIODevice::read (QNetworkReplyHttpImpl): device not open
09-25 16:31:40:420 30084 OCC::CheckServerJob::finished: error: status.php replied  0 ""
09-25 16:31:40:421 30084 unknown: QIODevice::read (QNetworkReplyHttpImpl): device not open
09-25 16:31:40:422 30084 OCC::ConnectionValidator::slotNoStatusFound: void OCC::ConnectionValidator::slotNoStatusFound(QNetworkReply*) QNetworkReply::NetworkError(OperationCanceledError) "Operation canceled" ""
09-25 16:31:40:423 30084 OCC::AccountSettings::showConnectionLabel: "No connection to Nextcloud at <a href=\"https://my.domain2.com/cloud\">https://my.domain2.com/cloud</a>.\nOperation canceled"
09-25 16:31:40:472 30084 OCC::FolderMan::slotAccountStateChanged: Account "me2@my.domain2.com" disconnected or paused, terminating or descheduling sync folders
09-25 16:31:40:474 30084 OCC::Account::resetNetworkAccessManager: Resetting QNAM
09-25 16:31:42:865 30084 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(OperationCanceledError) "Operation canceled" QVariant(Invalid)
09-25 16:31:42:866 30084 unknown: QIODevice::read (QNetworkReplyHttpImpl): device not open
09-25 16:31:42:868 30084 OCC::CheckServerJob::finished: error: status.php replied  0 ""
09-25 16:31:42:869 30084 unknown: QIODevice::read (QNetworkReplyHttpImpl): device not open
09-25 16:31:42:870 30084 OCC::ConnectionValidator::slotNoStatusFound: void OCC::ConnectionValidator::slotNoStatusFound(QNetworkReply*) QNetworkReply::NetworkError(OperationCanceledError) "Operation canceled" ""
09-25 16:31:42:872 30084 OCC::AccountSettings::showConnectionLabel: "No connection to Nextcloud at <a href=\"https://my.domain1.com/oc\">https://my.domain1.com/oc</a>.\nOperation canceled"
09-25 16:31:42:907 30084 OCC::FolderMan::slotAccountStateChanged: Account "me1@my.domain1.com" disconnected or paused, terminating or descheduling sync folders
09-25 16:31:42:911 30084 OCC::Account::resetNetworkAccessManager: Resetting QNAM
```
</details>

**Comment by camilasan** [comment]
Hi! Could you try the same with one of our daily builds with a test account?
https://download.nextcloud.com/desktop/daily/ Thanks!

**Comment by r2evans** [comment]
Unfortunately I'm not near a hotspot wifi that triggers it, this only occurs for me when I'm on travel. (Doing so locally would require that I check in to a hotel to get access through them.) I'll keep this in mind the next time I travel, though, and see if I can reproduce it.

**Comment by MinIsMin** [comment]
Can confirm it's still an issue with 2.5.2git (ppa).
When sync is paused the client also pushes all changes to server.

**Comment by camilasan** [comment]
Hi, Could you try to run the most recent desktop client and server to check if you still have this issue? The latest version can be seen by checking [https://nextcloud.com/install/#install-clients](https://nextcloud.com/install/#install-clients) and [https://nextcloud.com/install/#instructions-server](https://nextcloud.com/install/#instructions-server) respectively.

If you still have this problem with the latest desktop client, please re-open the issue.

Thanks!

**Comment by r2evans** [comment]
I'll test when I get the chance, thank you for the heads up.

Can you help me understand how the server has any say on the client connecting when it should not?

**Comment by r2evans** [comment]
(BTW, I had no idea that 20.0.7 was released. The [blog](https://nextcloud.com/blog/) has been silent on it. Perhaps a documentation bug?)

**Comment by camilasan** [comment]
> Can you help me understand how the server has any say on the client connecting when it should not?

If you stopped the syncing, the server doesn't have a say on that. But the client does more than syncing. Pausing the syncing won't stop it from retrieving notifications for instance.

**Comment by jospoortvliet** [comment]
> (BTW, I had no idea that 20.0.7 or even 20.0.6 had been released. The [blog](https://nextcloud.com/blog/) has been silent on it. Perhaps a documentation bug?)

They were very minor releases that were done outside the usual update cycle, so we stuck with updating the changelog...
